vendor/github.com/containers/image/v5/docker/docker_transport.go

```diff
@@ -141,6 +168,9 @@ func (ref dockerReference) NewImage(ctx context.Context, sys *types.SystemContex
 // NewImageSource returns a types.ImageSource for this reference.
 // The caller must call .Close() on the returned ImageSource.
 func (ref dockerReference) NewImageSource(ctx context.Context, sys *types.SystemContext) (types.ImageSource, error) {
+	if ref.isUnknownDigest {
+		return nil, fmt.Errorf("Docker reference without a tag or digest is not supported for NewImageSource")
+	}
```

Comment:
(This works; it would be a tiny bit cleaner in `newImageSource` — we could/should document that `dockerImageSource.{logical,physical}Ref` are `!isUnknownDigest`, and having the enforcement and the data structure definition close together would be a bit easier to follow.

(Alternatively, document that the caller of `newImageSource` must ensure `!isUnknownDigest`; that works but requires extra attention.))

---

Also, uh, not having the check in `_transport.go` means that I wouldn’t insist on having a unit test for that condition :)